### PR TITLE
Remove duplicated package "tzdata".

### PR DESCRIPTION
**Pull request by iwanami-lin**
# Purpose of pull request

There is no big difference betweein meta-debian and mete-debian-extended. 
So remove bb file and add bbappend to customize its recipe.

# Test
## How to test
Build test
```
$ echo "IMAGE_INSTALL_append += \" tzdata \"" >> conf/local.conf
$ bitbake core-image-minimal
```

```
$ bitbake -e tzdata | grep CONFFILES_tzdata-core
# $CONFFILES_tzdata-core [2 operations]
CONFFILES_tzdata-core="/etc/localtime /etc/timezone"
```
Second /etc/timezone is an addition.

## Test result
```
$ MACHINE=qemuarm64 runqemu nographic
```
**Testcase tzdata at qemu.**
```
# date
Thu Jun  4 01:39:29 UTC 2020
# cp /usr/share/zoneinfo/Asia/Tokyo /etc/localtime
# reboot

rebooting

# date
Thu Jun  4 10:40:49 JST 2020

Timezone from UTC to JST.
```

### CODE REVIEW

File: recipes-debian/tzdata/tzdata_debian.bbappend [hunk 1]
```diff
@@ -0,0 +1 @@
+CONFFILES_tzdata-core += "${@ "${sysconfdir}/timezone" if bb.utils.to_boolean(d.getVar('INSTALL_TIMEZONE_FILE')) else "" }"
```
Comment (Ponsuke07) [inline recipes-debian/tzdata/tzdata_debian.bbappend:2]: This line doesn't work.
**timezone** will be installed even if **INSTALL_TIMEZONE_FILE** set to **"0"**

```
$: echo 'IMSTALL_TIMEZONE_FILE_pn-tzdata = "0"' >> conf/local.conf
$: bitbake -e tzdata |grep CONFFILES_tzdata-core
#     "${@ "${sysconfdir}/timezone" if bb.utils.to_boolean(d.getVar('INSTALL_TIMEZONE_FILE')) else "" }"
# $CONFFILES_tzdata-core [2 operations]
#     "${@ "${sysconfdir}/timezone" if bb.utils.to_boolean(d.getVar('INSTALL_TIMEZONE_FILE')) else "" }"
#   "${sysconfdir}/localtime ${sysconfdir}/timezone ${@ "${sysconfdir}/timezone" if bb.utils.to_boolean(d.getVar('INSTALL_TIMEZONE_FILE')) else "" }"
CONFFILES_tzdata-core="/etc/localtime /etc/timezone "
# $INSTALL_TIMEZONE_FILE [2 operations]
INSTALL_TIMEZONE_FILE="0"
# $INSTALL_TIMEZONE_FILE_pn-tzdata
INSTALL_TIMEZONE_FILE_pn-tzdata="0"
```

Comment (Ponsuke07) [reply recipes-debian/tzdata/tzdata_debian.bbappend:2]: Also, "${sysconfdir}}/timezone" is in FIELS_tzdata-core on meta-debian.
If INSTALL_TIMEZONE_FILE is "0" then you may need to strip it .

**Review by Ponsuke07 — CHANGES_REQUESTED** [review]

File: recipes-debian/tzdata/tzdata_debian.bbappend (hunk 1 above)
Comment (hiraku-wfs) [reply recipes-debian/tzdata/tzdata_debian.bbappend:2]: Can we overwrite the meta-debian's CONFFILES in bbappend in the following manner?

```
CONFFILES_tzdata-core = "${sysconfdir}/localtime"
CONFFILES_tzdata-core += "${@ "${sysconfdir}/timezone" if bb.utils.to_boolean(d.getVar('INSTALL_TIMEZONE_FILE')) else "" }"
```

Comment (Ponsuke07) [reply recipes-debian/tzdata/tzdata_debian.bbappend:2]: > Can we overwrite the meta-debian's CONFFILES in bbappend in the following manner?

Alternatively, there is the following manner. (I'm fine with either.)
```
CONFFILES_tzdata-core_remove = "${@ "${sysconfdir}/timezone" if not bb.utils.to_boolean(d.getVar('INSTALL_TIMEZONE_FILE')) else "" }"
```

Comment (hiraku-wfs) [reply recipes-debian/tzdata/tzdata_debian.bbappend:2]: Once we use the "remove", we can not override it in further bbappend. So I prefer the former one.

Comment (iwanami-lin) [reply recipes-debian/tzdata/tzdata_debian.bbappend:2]: I fixed this.
~~~
CONFFILES_tzdata-core += "${@ "${sysconfdir}/timezone" if bb.utils.to_boolean(d.getVar('INSTALL_TIMEZONE_FILE')) else "" }"
~~~
~~~
$ bitbake -e tzdata | grep CONFFILES_tzdata-core
# $CONFFILES_tzdata-core [3 operations]
CONFFILES_tzdata-core="/etc/localtime /etc/timezone /etc/timezone"
~~~
↓
~~~
CONFFILES_tzdata-core = "${sysconfdir}/localtime"
CONFFILES_tzdata-core += "${@ "${sysconfdir}/timezone" if bb.utils.to_boolean(d.getVar('INSTALL_TIMEZONE_FILE')) else "" }"
~~~

~~~
$ bitbake -e tzdata | grep CONFFILES_tzdata-core
# $CONFFILES_tzdata-core [3 operations]
CONFFILES_tzdata-core="/etc/localtime /etc/timezone"
~~~

Comment (hiraku-wfs) [reply recipes-debian/tzdata/tzdata_debian.bbappend:2]: How about the result when the INSTALL_TIMEZONE_FILE is 0?

Comment (iwanami-lin) [reply recipes-debian/tzdata/tzdata_debian.bbappend:2]: $ echo 'INSTALL_TIMEZONE_FILE_pn-tzdata = "0"' >> conf/local.conf

$ bitbake -e tzdata | grep CONFFILES_tzdata-core
CONFFILES_tzdata-core="  /etc/localtime"

**Comment by hiraku-wfs** [comment]
Thanks!

**Pull request merged**